src/shared/condition.c

```diff
@@ -611,7 +611,7 @@ static int condition_test_needs_update(Condition *c) {
                         return true;
                 }
 
-                timespec_store(&other.st_mtim, timestamp);
+                return timespec_load_nsec(&usr.st_mtim) > timestamp;
```

Comment:
Hmm, so maybe invert the condition above and immediately return if
`usr.st_mtim.tv_nsec == 0 || other.st_mtim.tv_nsec > 0`?